videoprocess/vacopy.cpp

```diff
@@ -561,25 +559,15 @@ vpp_context_create()
     va_status = vaInitialize(va_dpy, &major_ver, &minor_ver);
     assert(va_status == VA_STATUS_SUCCESS);
 
-    /* Check whether VPP is supported by driver */
-    VAEntrypoint entrypoints[5];
-    int32_t num_entrypoints;
-    num_entrypoints = vaMaxNumEntrypoints(va_dpy);
-    va_status = vaQueryConfigEntrypoints(va_dpy,
-                                         VAProfileNone,
-                                         entrypoints,
-                                         &num_entrypoints);
-    CHECK_VASTATUS(va_status, "vaQueryConfigEntrypoints");
-
-    for (j = 0; j < num_entrypoints; j++) {
-        if (entrypoints[j] == VAEntrypointVideoProc)
-            break;
-    }
-
-    if (j == num_entrypoints) {
-        printf("VPP is not supported by driver\n");
-        assert(0);
-    }
+    /* Check whether vaCopy is supported by driver */;
+    VADisplayAttribute va_disp_attr;
+    va_disp_attr.type = VADisplayAttribCopy;
+    va_status = vaGetDisplayAttributes(va_dpy, &va_disp_attr, 1);
+    
+    CHECK_VASTATUS(va_status, "vaGetDisplayAttrib");
+    if(va_disp_attr.value == VA_ATTRIB_NOT_SUPPORTED || va_disp_attr.value == 0)
```

Comment:
at line 540 we do:
option.bits.va_copy_mode = g_copy_method; // VA_COPY_MODE_BALANCE;
It would be better :
1) to make it parametrized via cmd line:
2) to check if the specified mode (even if it's hardcoded) matches attributes returned from vaGetDisplayAttributes